### PR TITLE
Handle commit error gracefully

### LINKED ISSUE #254: Commit has misleading error when creds are required but not supplied
Guohua Ouyang reports the below.  He used a private repository and neglected to include the creds command and appropriate un/pwd.  He received a 401 error and might better have been specified as "authentication failed" or some such.

[root@fedora tasks]# buildah commit nginx docker://docker.io/gouyang/nginx
Getting image source signatures
Skipping fetch of repeat blob sha256:ee1a8002360e897670d7edcc6bfde440f064aa014793e7e7a5dc049a01459299
Skipping fetch of repeat blob sha256:12c9ee9bfbc78f3c69092d8aed395fc85db739c90fb8f128c7896c1a03102fdb
Skipping fetch of repeat blob sha256:8481764ad3d61d907eceeaad333bb20b0da6c28c4420e6cfc712aee24f82b5ae
Copying blob sha256:4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1
 0 B / 32 B [------------------------------------------------------------------]
ERRO[0023] error committing container "nginx" to "docker://docker.io/gouyang/nginx": error copying layers and metadata: Error writing blob: Error initiating layer upload to /v2/gouyang/nginx/blobs/uploads/, status 401

[root@fedora tasks]# buildah commit --creds gouyang:xxxxxx nginx docker://docker.io/gouyang/nginx
Getting image source signatures
Skipping fetch of repeat blob sha256:ee1a8002360e897670d7edcc6bfde440f064aa014793e7e7a5dc049a01459299
Skipping fetch of repeat blob sha256:12c9ee9bfbc78f3c69092d8aed395fc85db739c90fb8f128c7896c1a03102fdb
Skipping fetch of repeat blob sha256:8481764ad3d61d907eceeaad333bb20b0da6c28c4420e6cfc712aee24f82b5ae
Copying blob sha256:4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1
 32 B / 32 B [=================================================================]
Copying config sha256:c47d98715367c9037b1aedd237e2ffe4c3b909864761c002951a16aece749f89
 4.87 KiB / 4.87 KiB [=========================================================]
Writing manifest to image destination
Storing signatures

### PR DESCRIPTION
**Pull request by ripcurld0**
This change gives a better error message when the commit fails because of bad authentication.

This fixes #254 and should be merged only after https://github.com/containers/image/pull/390 is merged.

### CODE REVIEW

File: cmd/buildah/commit.go [hunk 1]
```diff
@@ -143,7 +144,18 @@ func commitCmd(c *cli.Context) error {
 	}
 	err = builder.Commit(dest, options)
 	if err != nil {
-		return errors.Wrapf(err, "error committing container %q to %q", builder.Container, image)
+		switch nErr := errors.Cause(err).(type) {
+		case errcode.Errors:
+			oErr := nErr[0]
+			for _, e := range nErr[1:] {
+				oErr = errors.Wrapf(e, oErr.Error())
+			}
```
Comment (nalind) [inline cmd/buildah/commit.go]: Iteratively wrapping each error with a separate call to `errors.Wrapf()` is going to produce duplication in the call stack.  Perhaps converting the `errcode.Errors` into a `github.com/urfave/cli/MultiError` with its `NewMultiError()` function and then wrapping that result by calling `errors.Wrapf()` once would produce something more readable?

Comment (ripcurld0) [reply cmd/buildah/commit.go]: I found out that `errors.Wrapf`ing `MultiError` makes the error message a little bit confusing so I went with returning just `MultiError` instead.

Comment (nalind) [reply cmd/buildah/commit.go]: Ah, I was wondering about that.  Makes sense.

**Comment by rhatdan** [comment]
@ripcurld0 You have to sign your PR.

**Comment by rhatdan** [comment]
LGTM

**Comment by rhatdan** [comment]
@nalind @TomSweeneyRedHat PTAL

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Comment by rhatdan** [comment]
This should not have been merged yet.

**Comment by nalind** [comment]
Until https://github.com/containers/image/pull/390 gets merged (and vendored here), this should just produce the same behavior we have now, so no harm done, right?

**Comment by ripcurld0** [comment]
@nalind correct.

**Comment by nalind** [comment]
Thanks!

**Comment by ripcurld0** [comment]
@nalind no problem 👍

**Pull request closed without merging**